packages/apollo-datasource-rest/src/RESTDataSource.ts

```diff
@@ -104,9 +104,10 @@ export abstract class RESTDataSource<TContext = any> extends DataSource {
   protected parseBody(response: Response): Promise<object | string> {
     const contentType = response.headers.get('Content-Type');
     if (
-      contentType &&
-      (contentType.startsWith('application/json') ||
-        contentType.startsWith('application/hal+json'))
+      response.status !== 204 ||
```

Comment:
Are you sure you didn't mean to put this as an `&&` inside the parenthesis that contain `contentType.startsWith(...) || ...`?

The failing tests are because of `Content-type: plain` and you're _always_ short-circuiting when the `response.status !== 204`, which would be most all of the time, no?

Reply:
this is embarrassing..., it, indeed, should have been `&&` instead of `||`, it is not necessary to put it inside the parenthesis because it should pass all three conditions.

Thank you very much!

Reply:
No need to be embarrassed.  That's what failing tests and code-reviews are meant to help with. :smile:

Thank you for the first-time contribution!